nsds_lab_to_nwb/common/io.py

```diff
@@ -20,11 +24,14 @@ def write_yaml(yml_path, data, access_mode='w', default_flow_style=False, sort_k
     od_representer = (lambda dumper, data:
                       dumper.represent_mapping('tag:yaml.org,2002:map', data.items()))
     yaml.add_representer(OrderedDict, od_representer)
-    with io.open(yml_path, access_mode) as fh:
-        yaml.dump(data, fh,
-                  Dumper=MyDumper,
-                  default_flow_style=default_flow_style,
-                  sort_keys=sort_keys)
+    try:
+        with io.open(yml_path, access_mode) as fh:
+            yaml.dump(data, fh,
+                      Dumper=MyDumper,
+                      default_flow_style=default_flow_style,
+                      sort_keys=sort_keys)
+    except PermissionError:
+        logger.debug(f'Failed to write yaml due to permissions error in {yml_path}')
```

Comment:
I decided to implement the try except in write_yaml because I thought we wanted this behavior anytime write_yaml is called (since the user may not have write privileges). I could implement in outside of write_yaml, but then I would have to try-except the 6 or so references of write_yaml in ExpNoteReader and MetaDataReader and I thought it was simpler / more elegant. If you feel strongly about no logging in io.py, I can implement the try except in ExpNoteReader and MetaDataReader

Reply:
Thanks for the explanation, that sounds reasonable!